### PR TITLE
feat(api): PartialUpdateHostTemplate endpoint: handle macros (#1551)

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Feature PR contains #1551 

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

### CODE REVIEW

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -17801,3 +17801,12 @@ msgstr "Erreur lors de la liaison de catégories d'hôte à un modèle d'hôte"
 
 msgid "The %s does not exist with ID(s) '%s'"
 msgstr "Le champ %s n'existe pas avec le(s) ID(s) '%s'"
+
+msgid "Error while partially updating a host template"
+msgstr "Erreur lors de la correction d'un modèle d'hôte"
```
Comment (a-launois) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17806]:
```suggestion
msgstr "Erreur lors de la modification partielle d'un modèle d'hôte"
```

**Review by cg-tw — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

File: centreon/src/Core/CommandMacro/Domain/Model/CommandMacro.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    public function __construct(
+        private readonly int $commandId,
+        private readonly CommandMacroType $type,
+        private string $name,
+    ) {
+        $this->shortName = (new \ReflectionClass($this))->getShortName();
+
+        Assertion::positiveInt($commandId, "{$this->shortName}::commandId");
+
+        Assertion::notEmptyString($this->name, "{$this->shortName}::name");
```
Comment (callapa) [inline centreon/src/Core/CommandMacro/Domain/Model/CommandMacro.php:56]: No trim for the property **name** ?

File: centreon/src/Core/Host/Domain/Model/HostInheritance.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        $inheritanceLine = [];
+
+        $directParents = array_filter($parents, (fn($row) => $row['child_id'] === $hostTemplateId));
+
+        foreach ($directParents as $parent) {
+            $inheritanceLine = array_merge(
+                $inheritanceLine,
+                self::findInheritanceLine($parent['parent_id'], $parents)
+            );
+        }
```
Comment (callapa) [inline centreon/src/Core/Host/Domain/Model/HostInheritance.php:51]: Is it possible to have an infinite loop?

Comment (a-launois) [reply centreon/src/Core/Host/Domain/Model/HostInheritance.php:51]: No, circular inheritance is forbidden and will be check at template creation/update

File: centreon/src/Core/CommandMacro/Domain/Model/CommandMacro.php (hunk 2 above)
Comment (a-launois) [reply centreon/src/Core/CommandMacro/Domain/Model/CommandMacro.php:56]: No, as command macro cannot be created via API at the moment and their format cannot contains space when created via the UI

File: centreon/src/Core/HostMacro/Domain/Model/HostMacro.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    public function __construct(
+        private readonly int $hostId,
+        private string $name,
+        private string $value,
+    ) {
+        $this->shortName = (new \ReflectionClass($this))->getShortName();
+
+        Assertion::positiveInt($hostId, "{$this->shortName}::hostId");
+
+        $this->name = mb_strtoupper($name);
```
[the remark this answers is not in the document]
Comment (a-launois) [reply centreon/src/Core/HostMacro/Domain/Model/HostMacro.php:58]: UI behavior currently accepts and does not trim macro names

**Review by callapa — APPROVED** [review]

**Pull request merged**